### DISCUSSION TITLE
eslint-plugin-next no-html-link-for-pages does not work with workspaces, searches in wrong path

### DISCUSSION THREAD

**Discussion by Yzrsah** [vercel/next.js#24254] [Help]
When using workspaces, `@next/next/no-html-link-for-pages` searches for pages in the wrong directory. It searches for `theTopLevelProject/pages` instead of `theTopLevelProject/packages/someWorkspace/pages`. It does not check the path of the package and instead uses the path of the `node_modules`, which will not work with workspaces hoisting.

I.e.:
`Error while loading rule '@next/next/no-html-link-for-pages': Pages directory cannot be found at /Users/foo/theTopLevelProject/pages or /Users/foo/theTopLevelProject/src/pages. If using a custom path, please configure with the no-html-link-for-pages rule
`

Environment:
- Next 10.1.3
- Yarn V2 (nodeLinker: node-modules)

**Comment by jamesarosen** [comment]
https://nextjs.org/docs/messages/no-html-link-for-pages has no information about how to configure a custom path. The [unit test for the `no-html-link-for-pages` rule](https://github.com/vercel/next.js/blob/e783b0a2e85095b03a4eace331a5d85ed2521723/test/eslint-plugin-next/no-html-link-for-pages.unit.test.js#L10) suggests
```js
// .eslintrc.js
rules: {
  'no-html-link-for-pages': [2, path.join(__dirname, 'custom-pages')],
},
```
In your app, that has to be
```js
// .eslintrc.js
rules: {
  '@next/next/no-html-link-for-pages': [2, path.join(__dirname, 'custom-pages')],
},
```

**Reply by xenostar** [reply]
Oh man, thank you. This was driving me crazy. The documentation on how to configure this definitely needs to be added.

**Reply by convers39** [reply]
you saved my day!!!

**Comment by jakeorr** [comment]
I'm facing the same issue. It would be great if there was a way to support a monorepo setup.

**Comment by amaankulshreshtha** [comment]
I tried using the solution from this [link](https://nextjs.org/docs/basic-features/eslint#rootdir). But it still gives the same error. Below is my code.

```js
module.exports = {
  root: true,
  extends: ['airbnb-base', 'next', '@react-native-community'],
  rules: {
    'comma-dangle': ['error', 'never'],
    'no-console': 'error',
    'no-restricted-syntax': ['error', 'ForInStatement', 'LabeledStatement', 'WithStatement'],
    'no-param-reassign': [
      'error',
      {
        props: true,
        ignorePropertyModificationsFor: ['state', 'acc']
      }
    ]
  },
  settings: {
    next: {
      rootDir: './packages/web/'
    }
  },
  ignorePatterns: ['dist', 'node_modules']
};
```

**Reply by hanselabreu** [reply]
I tried this as well and no luck

**Reply by naeemba** [reply]
This worked for me:
```
    'next': {
      rootDir: 'packages/*/',
    },
```

**Comment by sintijab** [comment]
You need to add plugin:@next/next/recommended to config

**Comment by thedaviddias** [comment]
Facing the same issue on a monorepo using Turborepo. I followed the doc but still getting the issue.

**Comment by siiron** [comment]
I have this problem in a monorepo using NX 20

**Comment by hizikNewton** [comment]
For turbo monorepo, this work for me 
include the rootDir of your next application in the .eslintrc.js

```
 settings: {
    "next": {
      "rootDir": "apps/web/"
    }
  }
```
